controllers/common/util.go

```diff
@@ -367,3 +367,71 @@ func GetControlNs(r client.Reader) (controlNs string) {
 
 	return
 }
+
+// ValidateCsMaps checks common-service-maps has no scope overlapping
+func ValidateCsMaps(cm *corev1.ConfigMap) error {
+	commonServiceMaps, ok := cm.Data["common-service-maps.yaml"]
+	if !ok {
+		return fmt.Errorf("there is no common-service-maps.yaml in configmap kube-public/common-service-maps")
+	}
+
+	var cmData CsMaps
+	if err := utilyaml.Unmarshal([]byte(commonServiceMaps), &cmData); err != nil {
+		return fmt.Errorf("failed to fetch data of configmap common-service-maps: %v", err)
+	}
+
+	nsSet := make(map[string]interface{})
+
+	for _, nsMapping := range cmData.NsMappingList {
+		// validate masterNamespace and controlNamespace
+		if cmData.ControlNs == nsMapping.CsNs {
+			return fmt.Errorf("invalid controlNamespace: %v", cmData.ControlNs)
+		}
+		if _, ok := nsSet[nsMapping.CsNs]; ok {
+			return fmt.Errorf("invalid map-to-common-service-namespace: %v", nsMapping.CsNs)
+		}
+		nsSet[nsMapping.CsNs] = struct{}{}
+		// validate CloudPak Namespace and controlNamespace
+		for _, ns := range nsMapping.RequestNs {
+			if cmData.ControlNs == ns {
+				return fmt.Errorf("invalid controlNamespace: %v", cmData.ControlNs)
+			}
+			if _, ok := nsSet[ns]; ok {
+				return fmt.Errorf("invalid requested-from-namespace: %v", ns)
+			}
+			nsSet[ns] = struct{}{}
```

Comment:
@Daniel-Fan 

My concern is how about users what to install common services in the same namespace as the cloudpaks. This function looks can't fulfill that.

Could we use a set for `CsNs` and another set for `RequestNs`？

Reply:
Great point, I did not consider this case at the first thought. Updated.